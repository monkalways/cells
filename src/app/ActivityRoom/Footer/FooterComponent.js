import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Button, Grid, withStyles,
} from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

import ScanCardDialog from '../../common/ScanCardDialog';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  onSignIn: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

const FooterComponent = ({ classes, isAuthenticated, onSignIn }) => (
  <div className={classes.root}>
    <AppBar position="static" className={classes.appBar}>
      <Grid
        container
        justify="center"
        alignItems="center"
        className={classes.container}
      >
        {!isAuthenticated && (
          <React.Fragment>
            <Button
              variant="contained"
              color="secondary"
              className={classes.button}
              onClick={() => onSignIn()}
            >
              Scan ID Card to Access
            </Button>
            <ScanCardDialog />
          </React.Fragment>
        )}
      </Grid>
    </AppBar>
  </div>
);

FooterComponent.propTypes = propTypes;

export default compose(
  withStyles((theme) => ({
    root: {
      flexGrow: 1,
    },
    appBar: {
      backgroundColor: theme.palette.background.default,
      height: 89,
    },
    button: {
      height: theme.spacing.unit * 7,
      width: theme.spacing.unit * 40,
      margin: theme.spacing.unit * 2,
    },
  })),
  withRouter,
)(FooterComponent);
