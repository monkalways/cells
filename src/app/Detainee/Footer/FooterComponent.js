import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Button, Grid, withStyles,
} from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

const FooterComponent = ({ classes, history }) => {
  const handleBackButtonClick = () => {
    history.push('/cells/A1/home');
  };

  return (
    <div className={classes.root}>
      <AppBar position="static" className={classes.appBar}>
        <Grid
          container
          alignItems="center"
          spacing={8}
          className={classes.container}
        >
          <Grid item sm>
            <Button
              variant="contained"
              size="small"
              className={classes.button}
              onClick={() => handleBackButtonClick()}
            >
              <ArrowBackIcon className={classes.icon} />
              Back
            </Button>
          </Grid>
        </Grid>
      </AppBar>
    </div>
  );
};

FooterComponent.propTypes = propTypes;

export default compose(
  withStyles((theme) => ({
    root: {
      flexGrow: 1,
    },
    container: {
      padding: theme.spacing.unit * 1,
    },
    icon: {
      fontSize: theme.typography.h3.fontSize,
    },
    button: {
      padding: theme.spacing.unit * 1,
      textAlign: 'center',
      color: theme.palette.text.secondary,
    },
    appBar: {
      backgroundColor: '#A8C6FA', // TODO: move color to theme
    },
    // navigation: {
    //   height: '100%',
    //   width: '100%',
    //   backgroundColor: '#A8C6FA', // TODO: move color to theme
    // },
  })),
  withRouter,
)(FooterComponent);
