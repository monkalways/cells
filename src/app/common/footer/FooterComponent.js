import React from "react";
import FooterThreeButtons from "./Footer.ThreeButtons";
import FooterToggle from "./Footer.Toggle";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";

const styles = {
  outerDiv: {
    display: "flex",
    justifyContent: "space-between",
    //padding: "0px 0px",
    width: 600,
    height: "9%",
    margin: 0,
    bottom: 0
  },
  footerButton: {
    height: 67,
    width: 67,
  },
  divScanButton: {
    display: "block",
    height: "50%",
    border: "3px solid red",
    margin: "auto",
    padding: "0px 5px",
    backgroundColor: "bisque",
    fontSize: 30,
    fontWeight: 600,
    bottom: 0
  }
}

const FooterComponent = (props) => {
  const { isAuthenticated, isUsedOn, isMeal, isMedication, isCellCheck } = props;
  const providedWelfare = isMeal || isMedication || isCellCheck;

  const primaryColor = isCellCheck ? "#e08823" : isMeal ? "#8bc349" : isMedication ? "#0000ff" : "#000000";
  const secondaryColor = isCellCheck ? "#485559" : "#ad144f";

  if (!isAuthenticated) {
    return (
      <div style={styles.outerDiv}>
        <div style={styles.divScanButton}>
          Scan ID Card to Access
        </div>
      </div>
    );
  }

  return (
    <div style={styles.outerDiv}>
      {isUsedOn === "cell" && !providedWelfare && <FooterThreeButtons />}
      {((isUsedOn === "cell" && providedWelfare) || isUsedOn === "detaineeProfile") &&
        <Button variant="fab" aria-label="Back" onClick={props.navigateBack} style={styles.footerButton}>
          <Avatar src={require("images/BackArrow.png")} style={styles.footerButton} />
        </Button>
      }
      {isUsedOn === "cell" && providedWelfare &&
        <Button variant="fab" aria-label="Save" onClick={props.saveWelfareData} style={styles.footerButton}>
          <Avatar src={require("images/SaveButton.png")} style={styles.footerButton} />
        </Button>
      }
      {isUsedOn === "cell" && providedWelfare &&
        <FooterToggle
          isCellCheck={isCellCheck}
          primaryColor={primaryColor}
          secondaryColor={secondaryColor}
        />
      }
    </div>
  );
}

export default FooterComponent;