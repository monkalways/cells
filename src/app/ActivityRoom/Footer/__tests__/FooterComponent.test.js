import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '@material-ui/core';
import { FooterComponent } from '../FooterComponent';

describe('FooterComponent', () => {
  let push;
  let onSignIn;
  let isAuthenticated;

  beforeEach(() => {
    push = jest.fn();
    onSignIn = jest.fn();
    isAuthenticated = false;
  });

  const setup = () => {
    const classes = {
      root: {},
      appBar: {},
      button: {},
    };
    return shallow(<FooterComponent
      classes={classes}
      onSignIn={onSignIn}
      history={{ push }}
      isAuthenticated={isAuthenticated}
    />);
  };

  it('should not render Scan ID Card button when authenticated', () => {
    isAuthenticated = true;
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Button)).not.toExist();
  });

  it('should render Scan ID Card button when not authenticated', () => {
    isAuthenticated = false;
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Button)).toExist();
  });

  it('should call onSignIn() button when Scan ID Card button is clicked', () => {
    isAuthenticated = false;
    const wrapper = setup();

    expect(wrapper.find(Button)).toExist();
    wrapper.find(Button).simulate('click');
    expect(onSignIn).toBeCalled();
  });
});
