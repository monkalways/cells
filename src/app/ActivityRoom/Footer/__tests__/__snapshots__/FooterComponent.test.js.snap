// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should not render Scan ID Card button when authenticated 1`] = `
<div
  className=""
>
  <WithStyles(AppBar)
    className=""
    position="static"
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      justify="center"
    />
  </WithStyles(AppBar)>
</div>
`;

exports[`FooterComponent should render Scan ID Card button when not authenticated 1`] = `
<div
  className=""
>
  <WithStyles(AppBar)
    className=""
    position="static"
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      justify="center"
    >
      <WithStyles(Button)
        className=""
        color="secondary"
        onClick={[Function]}
        variant="contained"
      >
        Scan ID Card to Access
      </WithStyles(Button)>
      <Connect(WithStyles(ScanCardDialogComponent)) />
    </WithStyles(Grid)>
  </WithStyles(AppBar)>
</div>
`;
